subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/results/ScenarioBuildResultData.groovy

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.performance.results
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties
+import org.gradle.performance.measure.DataSeries
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+class ScenarioBuildResultData {
+    String scenarioName
+    String webUrl
+    String testFailure
+    boolean successful
+    List<ExecutionData> currentCommitExecutions = []
+    List<ExecutionData> recentExecutions = []
+
+    boolean isAboutToRegress() {
+        return executions.any { it.regressionPercentage > 0 && it.confidencePercentage > IndexPageGenerator.DANGEROUS_REGRESSION_CONFIDENCE_THRESHOLD }
+    }
+
+    boolean isBuildFailed() {
+        return !successful && currentCommitExecutions.empty
+    }
+
+    boolean isFromCache() {
+        return successful && currentCommitExecutions.empty
+    }
+
+    double getRegressionSortKey() {
+        return executions.empty ? Double.NEGATIVE_INFINITY : executions[0].confidencePercentage * executions[0].regressionPercentage
```

Comment:
Let's use the `sign` here:
```
return executions.empty ? Double.POSITIVE_INFINITY : executions[0].confidencePercentage * Math.signum(executions[0].regressionPercentage)
```

If we don't use `signum`, we will get a strange mix of confidence percentage and regression percentage.

Reply:
Ok, I misunderstood you and thought a combination of confidence and percentage.